src/common/Utilities.py

```diff
@@ -231,14 +231,16 @@ def show_3D_array\
     return 0
 
 
-def check_tolerance(expected, actual, abstol=0, reltol=1e-4):
+def check_tolerance(expected, actual, abstol=0, reltol=2e-3):
```

Comment:
intentional to decrease default tolerance in this PR?

Reply:
taken from #844

Reply:
ok. confusing to mix the 2 PRs (might generate a conflict) but fine for me.